### ISSUE TITLE
runtime: Top-level `extend T::Sig` is not supported

### ISSUE THREAD

**Issue by AaronC81** [sorbet/sorbet#923]
#### Input

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20strong%0Arequire%20'sorbet-runtime'%0Aextend%20T%3A%3ASig%0A%0Asig%20%7B%20params(a%3A%20Numeric%2C%20b%3A%20Numeric).returns%20Numeric%20%7D%0Adef%20division(a%2C%20b)%0A%20%20%20%20a%20%2F%20b%0Aend%0A%0Aputs%20division(4%2C%202))

```ruby
# typed: strong
require 'sorbet-runtime'
extend T::Sig

sig { params(a: Numeric, b: Numeric).returns Numeric }
def division(a, b)
    a / b
end

puts division(4, 2)
```

#### Observed output

```
No errors! Great job.
```

The issue is not with the typechecking itself, but instead with `sorbet-runtime`. Executing this file results in:

```
Traceback (most recent call last):
        3: from main.rb:5:in `<main>'
        2: from /var/lib/gems/2.5.0/gems/sorbet-runtime-0.4.4254/lib/types/sig.rb:24:in `sig'
        1: from /var/lib/gems/2.5.0/gems/sorbet-runtime-0.4.4254/lib/types/private/methods/_methods.rb:16:in `declare_sig'
/var/lib/gems/2.5.0/gems/sorbet-runtime-0.4.4254/lib/types/private/methods/_methods.rb:307:in `install_hooks': undefined method `singleton_class?' for main:Object (NoMethodError)
Did you mean?  singleton_class
```

_(Note that line 5 is the `sig` line.)_

I am using "Ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux-gnu]", on Ubuntu 18.04. The behaviour does not change whether I execute the file using Bundler (`bundle exec ruby main.rb`) or just with the standard Ruby interpreter directly (`ruby main.rb`).

#### Expected behavior

I would expect that the file executes successfully and prints 2.

**Comment by AaronC81** [comment]
It appears that the issue only arises when the method definitions are in the top-level of the file (i.e. not inside a module). This works fine:

```
# typed: strong
require 'sorbet-runtime'

module A
    extend T::Sig

    sig { params(a: Numeric, b: Numeric).returns Numeric }
    def self.division(a, b)
        a / b
    end
end

puts A.division(4, 2)
```

**Comment by jez** [comment]
This is a duplicate of #199 but I like the examples and extra information in this issue, so I'm closing the other one.

**Comment by jez** [comment]
Fixed in #3491
